schools_change_detection/schools_change_detection.py

```diff
@@ -0,0 +1,862 @@
+                    else:
+                        continue
+            else:
+                filtered_moe_schools[unfiltered_school_id] = unfiltered_school
+    return filtered_moe_schools
+
+
+################
+# Analyse Data #
+################
+
+
+# look at using dictionaries so don't need to loop through lists
+def compare_schools(
+    facilities: Dict,
+    moe_schools: Dict,
+) -> List[MOESchool]:
+    """
+    Script checks for:
+     - updates to facilities - attributes and geoms
+     - facilities that need deleting
+     - schools which need to be added to facilities
+
+    Args:
+        facilities (Dict): all facilities which are schools
+        moe_schools (Dict): filtered moe_schools
+
+    Returns:
+        List[MOESchool]: a list of schools which need to be added to facilities.
+    """
+    new_schools = {}
+    # check each facility exists in MOE data, and if it does, then does it need updating?
+    for facility_source_id, facility in facilities.items():
+        facility_results = {
+            "attributes": None,
+            "geom": None,
+            # "occupancy": None,
+            "remove": False,
+        }
+        moe_match = moe_schools.get(facility_source_id)
+
+        if moe_match:
+            facility_results = {
+                "attributes": facility == moe_match,
+                "geom": facility.check_distance_polygon(moe_match),
+                # "occupancy": facility.check_occupancy(moe_match),
+                "remove": False,
+            }
+        else:
+            facility_results["remove"] = True
+
+        facility.parse_check_results(facility_results, moe_match)
+
+    # check each MOE school to see if it exists in facilities.
+    # if it doesn't then it needs to be added.
+    for school_id, school in moe_schools.items():
+        new_school_bool = False
+
+        # Does it exist in facilities
+        facility_match = facilities.get(school_id)
+        if not facility_match:
+            new_school_bool = True
+            new_schools[school_id] = school
+
+        moe_results = {"add": new_school_bool}
+
+        school.parse_check_results(moe_results)
+    return new_schools
+
+
+###################
+# Main entrypoint #
+###################
+
+
+def main(
+    faclities_input_file: Path,
+    faclities_db_conn: str,
+    output_dir: Path,
+) -> None:
+    """
+    Takes facilities location as an input and path to output.
+    """
+    # Create output directory
+    logger.info("Creating output directory at %s", output_dir)
+    os.mkdir(output_dir)
+
+    # Load facilities
+    if SOURCE_TYPE == "file":
+        logger.info("Loading facilities from %s", faclities_input_file)
+        facilities = load_file_source(faclities_input_file)
+    elif SOURCE_TYPE == "db":
+        logger.info("Loading facilities from DB")
+        facilities = load_db_source(faclities_db_conn)
+
+    # Load MOE Schooles
+    logger.info("Accessing MOE API")
+    unfiltered_moe_schools = request_moe_api()
```

Comment:
It would be good to have an option to use a local file (json / xlsx / csv) because we might want to run a custom check that doesn't use the latest data on the schools website.  Also, if we add Hospitals to this process, the input would be xlsx.  File inputs may be required for other facility types in the future too.